File: app/src/main/java/com/moisesborges/tvaddict/shows/ShowsView.java
package com.moisesborges.tvaddict.shows;

import android.support.annotation.NonNull;

import com.moisesborges.tvaddict.models.Show;
import com.moisesborges.tvaddict.mvp.BaseView;

import java.util.List;

/**
 * Created by Moisés on 12/04/2017.
 */

public interface ShowsView extends BaseView {
    void displayProgress(boolean isLoading);

    void displayTvShows(@NonNull List<Show> shows);

    void displayError();

    void navigateToShowDetails(@NonNull Show show);
}

File: app/src/main/java/com/moisesborges/tvaddict/shows/ShowsView.kt
package com.moisesborges.tvaddict.shows

import com.moisesborges.tvaddict.models.Show
import com.moisesborges.tvaddict.mvp.BaseView

/**
 * Created by Moisés on 12/04/2017.
 */

interface ShowsView : BaseView {

    fun displayProgress(isLoading: Boolean)

    fun displayTvShows(shows: List<Show>)

    fun displayError()

    fun navigateToShowDetails(show: Show)

    fun setPage(page: Int)

    fun displayMoreTvShows(shows: List<Show>)
}
